django/library/views.py

```diff
@@ -171,14 +173,41 @@ def post(self, request, *args, **kwargs):
         return redirect(request.META.get("HTTP_REFERER", ""))
 
 
-class PeerReviewReviewerListView(mixins.ListModelMixin, viewsets.GenericViewSet):
-    queryset = MemberProfile.objects.all()
-    serializer_class = RelatedMemberProfileSerializer
+class ChangePeerReviewPermission(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if request.user.has_perm("library.change_peerreview"):
+            return True
+        raise DrfPermissionDenied
+
+
+class PeerReviewerDashboardView(PermissionRequiredMixin, ListView):
+    template_name = "library/review/reviewers.jinja"
+    model = PeerReviewer
+    permission_required = "library.change_peerreview"
+    context_object_name = "reviewers"
+    paginate_by = 15
 
     def get_queryset(self):
-        query = self.request.query_params.get("query", "")
-        results = PeerReview.objects.find_candidate_reviewers(query)
-        return results
+        return PeerReviewer.objects.all()
+
+
+class PeerReviewerFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        # FIXME: when there query, return the full list of reviewers
+        query = request.query_params.get("query", None)
+        if query is None:
+            return queryset
+        return get_search_queryset(query, queryset)
+
+
+class PeerReviewerViewSet(CommonViewSetMixin, NoDeleteViewSet):
+    queryset = PeerReviewer.objects.exclude(is_active=False)
+    serializer_class = PeerReviewerSerializer
+    permission_classes = (ChangePeerReviewPermission,)
```

Comment:
We'll need to allow a user to create/update a peerreviewer record for themselves. I think the way to do this would be overriding `perform_create()` and `perform_update()` here to do the permission checks rather than using the permission class